File: WinFormsApplication/Dto/GameEntryRequestDto.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace hltb.Dto
{
    internal class GameEntryRequestDto
    {
        public long id { get; set; }

        [JsonPropertyName("custom_title")]
        public string customTitle { get; set; }

        public string note { get; set; }
        public long score { get; set; }
        public string status { get; set; }

        [JsonPropertyName("date_completed")]
        public string dateCompleted { get; set; }
        public long time { get; set; }
        public string platform { get; set; }

        [JsonPropertyName("client_id")]
        public long userId { get; set; }

        [JsonPropertyName("game_id")]
        public long gameId { get; set; }

        public static GameEntryDtoBuilder builder() { return new GameEntryDtoBuilder(); }

        public class GameEntryDtoBuilder
        {
            private GameEntryRequestDto _result;
            public GameEntryDtoBuilder()
            {
                _result = new GameEntryRequestDto();
            }

            public GameEntryDtoBuilder id(long id)
            {
                _result.id = id;
                return this;
            }

            public GameEntryDtoBuilder gameId(long gameId)
            {
                _result.gameId = gameId;
                return this;
            }

            public GameEntryDtoBuilder userId(long userId)
            {
                _result.userId = userId;
                return this;
            }

            public GameEntryDtoBuilder customTitle(string title)
            {
                _result.customTitle = title;
                return this;
            }

            public GameEntryDtoBuilder dateCompleted(DateOnly date)
            {
                _result.dateCompleted = $"{date.Year}-{ string.Format("{0:D2}", date.Month)}-{date.Day}";
                return this;
            }

            public GameEntryDtoBuilder platform(string platform)
            {
                _result.platform = platform;
                return this;
            }

            public GameEntryDtoBuilder score(long score)
            {
                _result.score = score;
                return this;
            }

            public GameEntryDtoBuilder status(string status)
            {
                _result.status = status;
                return this;
            }

            public GameEntryDtoBuilder time(long time)
            {
                _result.time = time;
                return this;
            }

            public GameEntryDtoBuilder note(string note)
            {
                _result.note = note;
                return this;
            }

            public GameEntryRequestDto build()
            {
                return _result;
            }
        }
    }
}

File: server/common/src/main/java/com/sawwere/titlecounter/common/dto/game/GameEntryRequestDto.java
package com.sawwere.titlecounter.common.dto.game;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Schema(description = "Dto for request for game entry entity")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameEntryRequestDto {
    @Schema(accessMode = Schema.AccessMode.READ_ONLY)
    private Long id;

    @Schema(description = "Game entry custom title given by user. Default value equals to game title itself")
    @Size(min = 1, max = 64)
    @JsonProperty(value = "custom_title")
    private String customTitle;

    @Schema(description = "User's note on the game")
    @Size(max = 512)
    private String note;

    @Schema(description = "User's score")
    @Min(1)
    @Max(10)
    private Integer score;

    @Schema(description = "Status of the entry", example = "completed")
    @NotBlank
    private String status;

    @Schema(description = "Date of completion of the game", example = "2024-01-24")
    @JsonProperty(value = "date_completed")
    private LocalDate dateCompleted;

    @Schema(description = "User's time in the game")
    @Min(0)
    private Long time;

    @Schema(description = "Platform on which game is played", deprecated = true)
    @Max(64)
    private String platform;

    @NotNull
    @JsonProperty(value = "user_id")
    private Long userId;

    @NotNull
    @JsonProperty(value = "game_id")
    private Long gameId;
}
